### PR TITLE
item: update item/doc for new acquisition

### PR DESCRIPTION
**Pull request by zannkukai**
- Adapts the item JSON schema
    * 'new_acquisition' flag
    * 'acquisition_date' field
- Adapts the item CLI
- Updates the items/holdings data files
- Update document indexing process to add new acquisition data into the
ES document indexes. Based on these data, it is now pssible to retreive
new acquisition for an organisation/library/location.
- Creates a new document result filter. This filter is based on a
'nested' query including a term query (for owning level structure) and a
range query (for acquisition date).
- Adapts unitests

Co-Authored-by: Renaud Michotte <renaud.michotte@gmail.com>

## Why are you opening this PR?

https://tree.taiga.io/project/rero21-reroils/us/1539?milestone=269576

## How to test?

- `poetry run bootstrap` & `poetry run setup`
- You could play with a new filter into document search page : `new_acquisition=[from:to]` ::
    * `/global/search/documents?new_acquisition=2020-01-01:2020-06-01` (new acquisition between 01/01/2020 and 31/05/2020)
    * `/global/search/documents?new_acquisition=:2020-06-01&location=3` (new acquisition until  31/05/2020 for `location/3`)
    * `/global/search/documents?library=2&new_acquisition=` (new acquisition until sysdate for `library/2`)
   * Your can also perform dynamic request (ES is wonoderfull) : `https://localhost:5000/global/search/documents?q=&page=1&size=10&new_acquisition=now-6M%2Fd` --> last 6 months new acquisition on all database. The result depends of the query call.

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?

### CODE REVIEW

File: rero_ils/modules/items/api/record.py [hunk 1]
```diff
@@ -368,3 +370,15 @@ def get_note(self, note_type):
...
                  if note.get('type') == note_type]
         return next(iter(notes), None)
+
+    @property
+    def is_new_acquisition(self):
+        """Is this item should be considered as a new acquisition.
+
+        :return True if Item is a new acquisition, False otherwise
+        """
+        if self.get('acquisition_date', None):
```
Comment (BadrAly) [inline rero_ils/modules/items/api/record.py:380]: you can also do it like this:
```
acquisition_date = self.get('acquisition_date', None) 
if acquisition_date:
   return datetime.datetime.strptime(
               acquisition_date,  '%Y-%m-%d') < datetime.datetime.now()
```

File: rero_ils/modules/items/api/record.py [hunk 2]
```diff
@@ -368,3 +370,15 @@ def get_note(self, note_type):
         notes = [note.get('content') for note in self.notes
                  if note.get('type') == note_type]
         return next(iter(notes), None)
+
+    @property
+    def is_new_acquisition(self):
```
Comment (BadrAly) [inline rero_ils/modules/items/api/record.py:375]: I guess the item will be always considered `new_acquisition` until someone deletes the `acquisition_date` from the item record. as long as that is what is requested by the PO, no problem

File: rero_ils/modules/items/api/record.py (hunk 1 above)
Comment (zannkukai) [reply rero_ils/modules/items/api/record.py:380]: done

File: rero_ils/modules/items/api/record.py (hunk 2 above)
Comment (zannkukai) [reply rero_ils/modules/items/api/record.py:375]: I'm pretty sure that it's the behavior that PO expected. You can filter the the date interval using the `new_acquisition` filter when you search a document. By example for 'last 6 months acquisition' : `https://localhost:5000/global/search/documents?q=&page=1&size=10&new_acquisition=now-6M%2Fd` (ES is wonderfull !! )

File: rero_ils/modules/documents/listener.py [hunk 3]
```diff
@@ -55,10 +55,10 @@ def enrich_document_data(sender, json=None, record=None, index=None,
                     'library_pid': holding['library']['pid']
                 }
             }
-            # replace this by an ES query
-            es_items = list(ItemsSearch().filter(
-                'term', holding__pid=holding.pid
-            ).scan())
+            # ITEMS LINKED TO THE HOLDING
```
Comment (jma) [inline rero_ils/modules/documents/listener.py:58]: why  with capital letters?

File: rero_ils/query.py [hunk 4]
```diff
@@ -45,6 +45,65 @@ def inner(values):
     return inner
 
 
+def acquisition_filter():
```
Comment (jma) [inline rero_ils/query.py:48]: Excellent!

**Pull request merged**